receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -98,33 +98,72 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 
 	// Get the branch count (future branch data) for each repo and record the given metrics
 	var wg sync.WaitGroup
+	wg.Add(len(repos))
+	var mux sync.Mutex
 
 	for _, repo := range repos {
 		repo := repo
 		name := repo.Name
 		trunk := repo.DefaultBranchRef.Name
+		now := now
 
-		wg.Add(1)
 		go func() {
 			defer wg.Done()
 
-			count, err := ghs.getBranches(ctx, genClient, name, trunk)
+			branches, count, err := ghs.getBranches(ctx, genClient, name, trunk)
 			if err != nil {
-				ghs.logger.Sugar().Errorf("error getting branch count for repo %s", zap.Error(err), repo.Name)
+				ghs.logger.Sugar().Errorf("error getting branch count: %v", zap.Error(err))
 			}
+
+			// Create a mutual exclusion lock to prevent the recordDataPoint
+			// SetStartTimestamp call from having a nil pointer panic
+			mux.Lock()
 			ghs.mb.RecordGitRepositoryBranchCountDataPoint(now, int64(count), name)
 
+			// Iterate through the branches populating the Branch focused
+			// metrics
+			for _, branch := range branches {
+				// See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/gitproviderreceiver/internal/scraper/githubscraper/README.md#github-limitations
+				// for more information as to why we do not emit metrics for
+				// the default branch (trunk) nor any branch with no changes to
+				// it.
+				if branch.Name == branch.Repository.DefaultBranchRef.Name || branch.Compare.BehindBy == 0 {
```

Comment:
Why do you use the `BehindBy` and not `AheadBy` to find branches that have no changes? I'd think that if a branch has no changes, its `AheadBy` is equal to zero, not necessarily the `BehindBy`. Am I misunderstanding something?

Reply:
Oh I see below is a reference to the GitHub API limitations doc. I read it and I still don't understand why `BehindBy` and `AheadBy` are swapped, but I'll take your word for it. :sweat_smile:

Reply:
Branches get compared against trunk, and therefor GitHub responds with `trunk is behind your branch by` which is the same as saying `your branch is ahead trunk by`.